docs/topics/control-flow.md

```diff
@@ -50,71 +50,152 @@ val max = if (a > b) {
 If you're using `if` as an expression, for example, for returning its value or
 assigning it to a variable, the `else` branch is mandatory.
 
-## When expression
+## When expressions and statements
 
-`when` defines a conditional expression with multiple branches. It is similar to the `switch` statement in C-like languages.
-Its simple form looks like this.
+`when` is a conditional expression that executes code based on multiple possible values or conditions. It is
```

Comment:
Do you think it's feasible to change "executes" for "runs"? Or any other word to avoid "execute" as much as possible, as mentioned in JB style guide and other TW style and best practices guidelines.